pydantic/json_schema.py

```diff
@@ -1445,8 +1445,15 @@ def _update_class_schema(
                 f"model_config['json_schema_extra']={json_schema_extra} should be a dict, callable, or None"
             )
 
+        if self._should_mark_class_as_deprecated(cls):
+            json_schema['deprecated'] = True
+
         return json_schema
 
+    @staticmethod
+    def _should_mark_class_as_deprecated(cls) -> bool:
+        return hasattr(cls, '__deprecated__') and getattr(cls, '__deprecated__')
```

Comment:
getattr is enough but you can make it more explicit

Reply:
@PrettyWood Sure. Do you think `bool(getattr(cls, '__deprecated__'))` is fine?

Reply:
LGTM once we decide on the syntax for this line!

Reply:
I just made a remark regarding the code with `hasattr` + `getattr`.
I now took a bit of time to understand the context and reading https://peps.python.org/pep-0702/ it seems you only care about the presence of the field
I can decide to write

```
@deprecated("")
class A: ...
```

and I'll have `A.__deprecated__ = ""`, which will be falsy and still valid.

So in the end I guess the initial approach with just `return hasattr(cls, '__deprecated__')` is the best

Reply:
![Screenshot 2024-05-15 at 15 58 46](https://github.com/pydantic/pydantic/assets/18406791/63c1c171-6fdb-4673-8873-9f9a63dc472e)

Reply:
What about a `hasattr` and then a false check? @NeevCohen @PrettyWood

Reply:
+1. As I said earlier "So in the end I guess the initial approach with just return hasattr(cls, '__deprecated__') is the best"